source/traction_boundary_conditions/initial_lithostatic_pressure.cc

```diff
@@ -0,0 +1,441 @@
+        {
+          spherical_representative_point[1] *= degrees_to_radians;
+
+          AssertThrow(spherical_representative_point[1] >= gm->west_longitude() &&
+                      spherical_representative_point[1] <= gm->east_longitude(),
+                      ExcMessage("Longitude of representative point outside chunk domain."));
+
+          if (dim ==3)
+            {
+              spherical_representative_point[2] *= degrees_to_radians;
+
+              AssertThrow(spherical_representative_point[2] >= gm->south_latitude() &&
+                          spherical_representative_point[2] <= gm->north_latitude(),
+                          ExcMessage("Latitude of representative point outside chunk domain."));
+            }
+
+          // set outer radius
+          spherical_representative_point[0] = gm->outer_radius();
+        }
+      else if (const GeometryModel::EllipsoidalChunk<dim> *gm = dynamic_cast<const GeometryModel::EllipsoidalChunk<dim>*> (&this->get_geometry_model()))
+        {
+          // If the eccentricity of the EllipsoidalChunk is non-zero, different boundaries will not have the
+          // same radius and calculating one pressure profile does not make sense
+          // Using the EllipsoidalChunk with eccentricity zero can be useful however,
+          // because the domain can be non-coordinate parallel.
+
+          AssertThrow(gm->get_eccentricity() == 0.0, ExcMessage("This plugin cannot be used with a non-zero eccentricity. "));
+
+          const std::vector<Point<2> > corners = gm->get_corners();
+
+          AssertThrow(spherical_representative_point[1] >= corners[1][1] &&
+                      spherical_representative_point[1] <= corners[0][1],
+                      ExcMessage("Longitude of representative point outside chunk domain."));
+
+          spherical_representative_point[1] *= degrees_to_radians;
+
+          if (dim ==3)
+            {
+
+              AssertThrow(spherical_representative_point[2] >= corners[2][2] &&
+                          spherical_representative_point[2] <= corners[1][2],
+                          ExcMessage("Latitude of representative point outside chunk domain."));
+
+              spherical_representative_point[2] *= degrees_to_radians;
+            }
+
+          spherical_representative_point[0] = gm->get_semi_major_axis_a();
+        }
+      else if (const GeometryModel::Sphere<dim> *gm = dynamic_cast<const GeometryModel::Sphere<dim>*> (&this->get_geometry_model()))
+        {
+          spherical_representative_point[0] = gm->radius();
+          spherical_representative_point[1] *= degrees_to_radians;
+          spherical_representative_point[2] *= degrees_to_radians;
+        }
+      // TODO: would be nice if we could combine Box and TwoMergedBoxes, but get_extents only exist for these
+      // two plugins, not all geometry models
```

Comment:
Excellent idea! It is cheap to test whether a point is inside if you have a domain that is described by a manifold, because you just have to evaluate the pull back. But the Triangulation doesn't know this, and so figuring out whether a point is inside is really expensive in general (O(number of cells), with a big constant).